presto-product-tests/src/main/java/com/facebook/presto/tests/convention/TestShowPartitions.java

```diff
@@ -76,5 +105,68 @@ public void testShowPartitionsFromHiveTable()
         String selectFromOnePartitionsSql = "show partitions from " + tableNameInDatabase;
         QueryResult partitionListResult = query(selectFromOnePartitionsSql);
         assertThat(partitionListResult).containsExactly(row(1), row(2));
+        assertColumnNames(partitionListResult, "part_col");
+    }
+
+    @Test(groups = {BASIC_SQL})
+    public void testShowPartitionsFromHiveTableWithTooManyPartitions()
+    {
+        String tableName = tablesState.get(PARTITIONED_TABLE_WITH_VARIABLE_PARTITIONS).getNameInDatabase();
+        createPartitions(tableName, TOO_MANY_PARTITIONS);
+
+        // Verify we created enough partitions for the test to be meaningful
+        assertExceptionThrownBy(() -> query("SELECT * FROM " + tableName))
+                .hasMessageMatching(".*: Query over table '\\S+' can potentially read more than \\d+ partitions");
+
+        QueryResult partitionListResult;
+
+        partitionListResult = query(format("SHOW PARTITIONS FROM %s WHERE part_col < 7", tableName));
+        assertThat(partitionListResult).containsExactly(row(0), row(1), row(2), row(3), row(4), row(5), row(6));
+        assertColumnNames(partitionListResult, "part_col");
+
+        partitionListResult = query(format("SHOW PARTITIONS FROM %s WHERE part_col < -10", tableName));
+        assertThat(partitionListResult).hasNoRows();
```

Comment:
can you add a test with just `SHOW PARTITIONS FROM %S` without `WHERE`

Reply:
Will be added in `Allow SHOW PARTITIONS for table above partition limit` commit, when it starts to work.